scripts/f5-cookie-decode.nse

```diff
@@ -0,0 +1,70 @@
+-- See here: https://support.f5.com/csp/article/K6917
+description = [[
+Decodes any unencrypted F5 BIG-IP cookies in the HTTP response.
+]]
+
+local http = require "http"
+local shortport = require "shortport"
+local stdnse = require "stdnse"
+local string = require "string"
+local utf8 = require "utf8"
+
+---
+-- @usage
+-- nmap -p <port> --script f5-cookie-decode <target>
+--
+-- @output
+-- PORT    STATE SERVICE
+-- 80/tcp  open  http
+-- | f5-cookie-decode:
+-- |_  BIGipServer<pool_name>:10.1.1.100:8080
+--
+-- @args f5-cookie-decode.path The URL path to request. The default path is "/".
+
+author = "Seth Jackson"
+license = "Same as Nmap--See https://nmap.org/book/man-legal.html"
+categories = { "discovery", "safe" }
+
+portrule = shortport.http
+
+action = function(host, port)
+  local path = stdnse.get_script_args(SCRIPT_NAME..".path") or "/"
+
+  local response = http.get(host, port, path)
+
+  if not response then
+    return
+  end
+
+  if not response.cookies then
+    return
+  end
+
+  local decoded = stdnse.output_table()
+
+  for _, cookie in pairs(response.cookies) do
+    if cookie.name:find("BIGipServer") then
+      local split = stdnse.strsplit("%.", cookie.value)
+
+      local host = tonumber(split[1])
+      local port = split[2]
+
+      if host then
+        local packed = string.pack("<I4", host)
+        local values = {}
+
+        for i = 1, #packed do
+          local c = packed:sub(i, i)
+          table.insert(values, utf8.codepoint(c))
+        end
+
+        local ip_address = stdnse.strjoin(".", values)
+        port = tonumber(stdnse.tohex(string.pack("<H", port)), 16)
```

Comment:
Similarly you could use `pack` - `unpack` to make it very clear what is going on here:
```lua
cport = (">I2"):unpack(("<I2"):pack(cport))